src/openrct2/actions/StaffHireNewAction.hpp

```diff
@@ -0,0 +1,347 @@
+            newPeep->path_check_optimisation = 0;
+            newPeep->type = PEEP_TYPE_STAFF;
+            newPeep->outside_of_park = 0;
+            newPeep->peep_flags = 0;
+            newPeep->paid_to_enter = 0;
+            newPeep->paid_on_rides = 0;
+            newPeep->paid_on_food = 0;
+            newPeep->paid_on_souvenirs = 0;
+            newPeep->staff_orders = _staffOrders;
+
+            uint16_t idSearchSpriteIndex;
+            Peep* idSearchPeep;
+
+            // We search for the first available id for a given staff type
+            uint32_t newStaffId = 0;
+            for (;;)
+            {
+                bool found = false;
+                ++newStaffId;
+
+                FOR_ALL_STAFF (idSearchSpriteIndex, idSearchPeep)
+                {
+                    if (idSearchPeep->staff_type != _staffType)
+                        continue;
+
+                    if (idSearchPeep->id == newStaffId)
+                    {
+                        found = true;
+                        break;
+                    }
+                }
+
+                if (!found)
+                    break;
+            }
+
+            newPeep->id = newStaffId;
+            newPeep->staff_type = _staffType;
+
+            PeepSpriteType spriteType = spriteTypes[_staffType];
+            if (_staffType == STAFF_TYPE_ENTERTAINER)
+            {
+                spriteType = static_cast<PeepSpriteType>(PEEP_SPRITE_TYPE_ENTERTAINER_PANDA + _entertainerType);
+            }
+            newPeep->name_string_idx = staffNames[_staffType];
+            newPeep->sprite_type = spriteType;
+
+            const rct_sprite_bounds* spriteBounds = g_peep_animation_entries[spriteType].sprite_bounds;
+            newPeep->sprite_width = spriteBounds->sprite_width;
+            newPeep->sprite_height_negative = spriteBounds->sprite_height_negative;
+            newPeep->sprite_height_positive = spriteBounds->sprite_height_positive;
+
+            if (_autoPosition == true)
+            {
+                AutoPositionNewStaff(newPeep);
+            }
+            else
+            {
+                // NOTE: This state is required for the window to act.
+                newPeep->state = PEEP_STATE_PICKED;
+
+                sprite_move(newPeep->x, newPeep->y, newPeep->z, (rct_sprite*)newPeep);
+                invalidate_sprite_2((rct_sprite*)newPeep);
+            }
+
+            // Staff uses this
+            newPeep->time_in_park = gDateMonthsElapsed;
+            newPeep->pathfind_goal.x = 0xFF;
+            newPeep->pathfind_goal.y = 0xFF;
+            newPeep->pathfind_goal.z = 0xFF;
+            newPeep->pathfind_goal.direction = 0xFF;
+
+            uint8_t colour = staff_get_colour(_staffType);
+            newPeep->tshirt_colour = colour;
+            newPeep->trousers_colour = colour;
+
+            // Staff energy determines their walking speed
+            newPeep->energy = 0x60;
```

Comment:
We could make this a constant. There is a PR that wants to make this a customisable value within cheats...

Reply:
Let's do so, then.